lib/cocoapods/generator/prepare_artifacts_script.rb

```diff
@@ -165,6 +166,17 @@ def script
             fi
 
             install_framework "$basepath/$target_path" "$embed"
+
+            if [[ -z "$dsym_folder" || ! -d "$dsym_folder" ]]; then
+              return
+            fi
+             
+            dsyms=($(ls "$dsym_folder"))
+              
+            local target_dsym=""
+            for i in ${!dsyms[@]}; do
+              install_artifact "$dsym_folder/${dsyms[$i]}" "$CONFIGURATION_BUILD_DIR" "true"
```

Comment:
in the prepare step, we now copy _all_ dsyms and filter out the ones we need inside the `Embed Frameworks` script